packages/gatsby-cli/src/reporter/redux/index.ts

```diff
@@ -11,6 +13,75 @@ let store = createStore(
   {}
 )
 
+function calculateTimeoutDelay(
+  envVarValue: string | undefined,
+  defaultValue: number,
+  min: number
+): number | null {
+  if (!envVarValue) {
+    return null
+  } else if (envVarValue === `1`) {
+    // toggling env vars with "1" is quite common - because we allow to set
+    // specific timeout values with env var, this special case is added
+    // (1ms timeout makes little sense - that's too low value to be used as-is)
+    return defaultValue
+  }
+
+  const parsedToNumber = parseInt(envVarValue)
+  if (isNaN(parsedToNumber)) {
+    // it's truthy, but not a number - let's enable it with default value
+    return defaultValue
+  }
+
+  // allow to custom specific timeout value, but also keep in mind that
+  // setting env var to "1" is common way of toggling features
+  // and we don't want to set to 1ms in those cases
+  return Math.max(parsedToNumber, min)
+}
+
+const stuckStatusDiagnosticTimeoutDelay = calculateTimeoutDelay(
+  process.env.GATSBY_DIAGNOSTIC_STUCK_STATUS_TIMEOUT,
+  1000 * 60 * 5, // 5 minutes default timeout
+  1000 * 5 // 5 seconds minimal value (this is mostly useful for debugging diagnostic code itself)
```

Comment:
Could do like `const FIVE_MINUTES = ` etc. Minor.

Reply:
Done as part of https://github.com/gatsbyjs/gatsby/pull/26618/commits/e10a55644f3f0dd8a4bd419c7fa23075660e3e93